BraintreeCore/src/main/java/com/braintreepayments/api/core/AnalyticsEventParams.kt

```diff
@@ -28,5 +28,5 @@ data class AnalyticsEventParams @JvmOverloads constructor(
     var endpoint: String? = null,
     val experiment: String? = null,
     val paymentMethodsDisplayed: List<String> = emptyList(),
-    val appSwitchUrl: String? = null
+    var appSwitchUrl: String? = null
```

Comment:
Is it possible to keep this property a `val`? We may need to move where this field is getting set before sending the analytic event, but I see some benefits in keeping these fields immutable. It prevents state from being held in multiple places and eliminates the chance of fields getting out of date.

Reply:
Wouldn't we want to make that change then across the board? The properties `payPalContextId`, `linkType`, `isVaultRequest`, `startTime`, `endTime`, and `endpoint` are all `var`s in `AnalyticsEventParams`. I'm not opposed to changing the pattern but was following what looked to be an already established pattern. If we'd like to move away from the pattern it seems like we should tackle all of `AnalyticsEventParams` as a whole vs this one property. Thoughts?

Reply:
Yep, all of the other properties in the events param class should ideally be `val`s, but I think this is existing tech debt. A good rule of thumb is that we should try to keep properties immutable if the parent data class is just used to pass data into another function.

Reply:
Yep, all of the other properties in the events param class should ideally be `val`s, but I think this is existing tech debt. A good rule of thumb is that we should try to keep properties immutable if the parent data class is just used to pass data into another function.

Reply:
Gotcha, yeah it is pretty confusing since that isn't the pattern we established here. In any case, update in commit https://github.com/braintree/braintree_android/pull/1226/commits/736fee1cb65e29498ee5937f03355e50ceea847a and created ticket DTMOBILES-1298 to address this tech debt.